src/data/list/basic.lean

```diff
@@ -4038,7 +4038,14 @@ by dsimp [Ico]; simp [nat.add_sub_cancel_left]
 theorem pred_singleton {m : ℕ} (h : m > 0) : Ico (m-1) m = [m-1] :=
 by dsimp [Ico]; rw nat.sub_sub_self h; simp
 
--- Someone put me out of my misery (no human suffering should be needed to prove this):
+theorem gt_nil {n m : ℕ} (h : n > m) : Ico n m = [] :=
```

Comment:
this holds more general for `n >= m`, also I would call it `eq_nil_of_le`

Reply:
Fixed, thanks.